datadog_lambda/tracing.py

```diff
@@ -156,27 +114,18 @@ def extract_context_from_lambda_context(lambda_context):
     dd_trace libraries inject this trace context on synchronous invocations
     """
     client_context = lambda_context.client_context
-    trace_id = None
-    parent_id = None
-    sampling_priority = None
     if client_context and client_context.custom:
         if "_datadog" in client_context.custom:
             # Legacy trace propagation dict
-            dd_data = client_context.custom.get("_datadog", {})
-            trace_id = dd_data.get(TraceHeader.TRACE_ID)
-            parent_id = dd_data.get(TraceHeader.PARENT_ID)
-            sampling_priority = dd_data.get(TraceHeader.SAMPLING_PRIORITY)
+            dd_data = client_context.custom.get("_datadog")
+            context = propagator.extract(dd_data)
         elif (
             TraceHeader.TRACE_ID in client_context.custom
             and TraceHeader.PARENT_ID in client_context.custom
             and TraceHeader.SAMPLING_PRIORITY in client_context.custom
         ):
-            # New trace propagation keys
-            trace_id = client_context.custom.get(TraceHeader.TRACE_ID)
-            parent_id = client_context.custom.get(TraceHeader.PARENT_ID)
-            sampling_priority = client_context.custom.get(TraceHeader.SAMPLING_PRIORITY)
-
-    return trace_id, parent_id, sampling_priority
+            context = propagator.extract(client_context.custom)
+        return context
```

Comment:
This is a bug. If neither the `if` or the `elif` are hit, then the variable `context` is undefined.